types/src/client.rs

```diff
@@ -148,6 +148,10 @@ pub enum FrontToBack {
 	// Such operations will be blocked until a response is received or the background
 	// thread has been terminated.
 	SubscriptionClosed(SubscriptionId),
+	/// Terminate background thread.
+	///
+	/// This implies that all pending operations are ignored and the connection is terminated.
+	Terminate,
```

Comment:
we need this message because once we have either pending request or subscriptions the background channel Sender is cloned and the drop impl won't work.

another option is keep it as it is and require users to explicitly drop subscriptions for this to work.